core/src/main/scala/com/salesforce/op/stages/impl/classification/OpXGBoostClassifier.scala

```diff
@@ -0,0 +1,187 @@
+
+  def setLambda(value: Double): this.type = set(lambda, value)
+
+  def setAlpha(value: Double): this.type = set(alpha, value)
+
+  def setTreeMethod(value: String): this.type = set(treeMethod, value)
+
+  def setGrowPolicy(value: String): this.type = set(growPolicy, value)
+
+  def setMaxBins(value: Int): this.type = set(maxBins, value)
+
+  def setSketchEps(value: Double): this.type = set(sketchEps, value)
+
+  def setScalePosWeight(value: Double): this.type = set(scalePosWeight, value)
+
+  def setSampleType(value: String): this.type = set(sampleType, value)
+
+  def setNormalizeType(value: String): this.type = set(normalizeType, value)
+
+  def setRateDrop(value: Double): this.type = set(rateDrop, value)
+
+  def setSkipDrop(value: Double): this.type = set(skipDrop, value)
+
+  def setLambdaBias(value: Double): this.type = set(lambdaBias, value)
+
+  // setters for learning params
+  def setObjective(value: String): this.type = set(objective, value)
+
+  def setBaseScore(value: Double): this.type = set(baseScore, value)
+
+  def setEvalMetric(value: String): this.type = set(evalMetric, value)
+
+  def setTrainTestRatio(value: Double): this.type = set(trainTestRatio, value)
+
+  def setNumEarlyStoppingRounds(value: Int): this.type = set(numEarlyStoppingRounds, value)
+
+  def setCustomObj(value: ObjectiveTrait): this.type = set(customObj, value)
+
+  def setCustomEval(value: EvalTrait): this.type = set(customEval, value)
+
+}
+
+
+/**
+ * Class that takes in a spark [[XGBoostClassificationModel]] and wraps it into an OP model which returns a
+ * Prediction feature
+ *
+ * @param sparkModel    model to wrap
+ * @param operationName unique name of the operation this stage performs
+ * @param uid           uid to give stage
+ */
+class OpXGBoostClassificationModel
+(
+  sparkModel: XGBoostClassificationModel,
+  uid: String = UID[OpXGBoostClassificationModel],
+  operationName: String = classOf[XGBoostClassifier].getSimpleName
+)(
+  implicit tti1: TypeTag[RealNN],
+  tti2: TypeTag[OPVector],
+  tto: TypeTag[Prediction],
+  ttov: TypeTag[Prediction#Value]
+) extends OpProbabilisticClassifierModel[XGBoostClassificationModel](
+  sparkModel = sparkModel, uid = uid, operationName = operationName
+) {
+  import OpXGBoost._
+
+  protected def predictRawMirror: MethodMirror = throw new NotImplementedError()
+  protected def raw2probabilityMirror: MethodMirror = throw new NotImplementedError()
+  @transient lazy val probability2predictionMirror =
+    reflectMethod(getSparkMlStage().get, "probability2prediction")
+
+  private lazy val model = getSparkMlStage().get
+  private lazy val booster = model.nativeBooster
+  private lazy val treeLimit = model.getTreeLimit.toInt
+
+  override def transformFn: (RealNN, OPVector) => Prediction = (label, features) => {
+    val data = removeMissingValues(Iterator(features.value.asXGB), model.getMissing)
+    val dm = new DMatrix(dataIter = data)
+    // TODO: can we avoid two booster.predict calls here?
```

Comment:
It shouldn't be hard to implement a method to convert raw predictions to probabilities. You think it's worth it?